Don't use global np.random.seed in tests

praths007 opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
Don't use global np.random.seed in tests #13345
<!--
Example: Fixes #13345  See also #3456. 
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.
Replaced instances in tests and doctests where seed was set globally using `np.random.seed` with 
`rng = np.random.RandomState`
so that the class variable `rng `can be used to generate random numbers instead of the global value of seed set using np.random.seed.

#### Any other comments?
Issue  #13345 can be checked for more details


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

praths007 committed:
initial commit

praths007 committed:
Merge branch 'master' of https://github.com/scikit-learn/scikit-learn into Don't_use_global_np.random.seed_in_tests

praths007 committed:
used random class

rth commented on sklearn/__init__.py:
```diff
@@ -96,13 +96,12 @@ def setup_module(module):
-    np.random.seed(_random_seed)
-    random.seed(_random_seed)
+    rng = np.random.RandomState(_random_seed)
+    rng.seed(_random_seed)
```
Hah, I was not aware this was happening. The `setup_module` function was last touched 4 years ago. Maybe let's keep it unchanged for now, and see in some other PR if we want to change this.

praths007 replied:
Thank You @rth .
I have reverted this bit and will be fixing places where the tests are breaking. (the tests might probably be failing because of using RandomState.random instead of RandomState.random_sample)

rth reviewed (commented):
Thanks for taking this on. Let's see if tests pass, otherwise looks good.

praths007 committed:
fixed failing testcases, reverted __init__.py

praths007 committed:
fixed failing testcases #2
- passed rng as parameter to ParameterSampler class
- changed seed from 0 to 42 (as original)

praths007 committed:
fixed failing testcases #2
- passed rng as parameter to SparseRandomProjection class

praths007 committed:
fixed failing testcases #4
- passed rng as parameter to GaussianRandomProjection class

praths007 committed:
fixed failing test case because of flake 8

praths007 commented:
@rth - all tests have passed now

rth reviewed (approved):
Thank you @praths007 !

agramfort commented:
thx @praths007
